### ISSUE TITLE
tsan/timer_list_test failure

### ISSUE THREAD

**Issue by dgquintas** [grpc/grpc#5297]
https://grpc-testing.appspot.com/job/gRPC_master/15215/config=tsan,language=c,platform=linux/testReport/(root)/tests/bins_tsan_timer_list_test/

```
D0218 00:44:15.876649252   18905 test_config.c:159]          test slowdown: machine=1.000000 build=2.000000 total=2.000000
E0218 00:44:15.917723965   18905 timer_list_test.c:143]      assertion failed: 1 == grpc_timer_check(&exec_ctx, tfm(2), NULL)
==================
WARNING: ThreadSanitizer: signal-unsafe call inside of a signal (pid=18905)
    #0 <null> <null>:0 (0x7e0000068981)
    #1 <null> <null>:0 (0x7e00000edfc0)
    #2 <null> <null>:0 (0x7e00000ede1a)
    #3 <null> <null>:0 (0x7e0000065209)
    #4 <null> <null>:0 (0x7e00000bfbc3)
    #5 <null> <null>:0 (0x7e00000c0028)
    #6 __libc_start_main /build/glibc-3Vu5mt/glibc-2.19/csu/libc-start.c:287 (libc.so.6+0x000000021b44)

SUMMARY: ThreadSanitizer: signal-unsafe call inside of a signal ??:0 ??
==================
Command exited with non-zero status 66
real 0.33
user 0.01
sys 0.24
```

**Comment by jtattermusch** [comment]
@ctiller , IIRC you did fix this one a few days ago?

**Comment by vjpai** [comment]
Fixed by #5662
